cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/aggregator/MetricSampleAggregator.java

```diff
@@ -188,6 +180,23 @@ public boolean addSample(PartitionMetricSample sample, boolean leaderValidation,
     return true;
   }
 
+  private void maybeEvictOldSnapshots() {
+    if (_windowedAggregatedPartitionMetrics.size() > _numSnapshotsToKeep + 1) {
+      synchronized (this) {
```

Comment:
Its cleaner to just have this method synchronized; one less block of code.

Reply:
Otherwise LGTM!

Reply:
We are not making the method synchronized to avoid unnecessary locking.